### PR TITLE
ISPN-10220 Create named lambdas to identify which test is running

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.jboss.org/browse/ISPN-10220

Now the lambda tests will be stored in the XML like:

```xml
  <testcase classname="org.infinispan.api.SimpleCacheTest" name="testLockedStreamActuallyLocks([functional-command, true])" time="0.001"/>
  <testcase classname="org.infinispan.api.SimpleCacheTest" name="testLockedStreamActuallyLocks([put-async, true])" time="0.001"/>
  <testcase classname="org.infinispan.api.SimpleCacheTest" name="testRetainAllMethodOfValuesCollection" time="0.001"/>
  <testcase classname="org.infinispan.api.SimpleCacheTest" name="testLockedStreamActuallyLocks([functional-command, false])" time="0.001"/>
```

If the test was not implemented with the named lambda we will have:

```xml
  <testcase classname="org.infinispan.api.SimpleCacheTest" name="testLockedStreamActuallyLocks([org.infinispan.api.APINonTxTest$$Lambda$266/0x00000008003df840@1fa88115, false])" time="0.001">
    <failure type="java.lang.IllegalStateException" message="Cannot identify what test is running">
      <![CDATA[java.lang.IllegalStateException: Cannot identify what test is running
at org.infinispan.commons.test.PolarionJUnitXMLReporter.deepToStringParameters(PolarionJUnitXMLReporter.java:388)
at org.infinispan.commons.test.PolarionJUnitXMLReporter.testName(PolarionJUnitXMLReporter.java:346)
at org.infinispan.commons.test.PolarionJUnitXMLReporter.createElement(PolarionJUnitXMLReporter.java:223)
at org.infinispan.commons.test.PolarionJUnitXMLReporter.createElementFromTestResults(PolarionJUnitXMLReporter.java:201)
at org.infinispan.commons.test.PolarionJUnitXMLReporter.generateReport(PolarionJUnitXMLReporter.java:188)
at org.infinispan.commons.test.PolarionJUnitXMLReporter.onFinish(PolarionJUnitXMLReporter.java:130)
... Removed 16 stack frames]]>
    </failure>
  </testcase> <!-- testLockedStreamActuallyLocks([org.infinispan.api.APINonTxTest$$Lambda$266/0x00000008003df840@1fa88115, false]) -->
```

### CODE REVIEW

**Comment by diegolovison** [comment]
This is a POC. If you agree with that, I am going to apply in all tests.

File: commons-test/src/main/java/org/infinispan/commons/lambda/NamedBiConsumer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.infinispan.commons.lambda;
+
+import java.util.function.BiConsumer;
+
+public class NamedBiConsumer<T, U> implements BiConsumer<T, U> {
```
Comment (danberindei) [inline commons-test/src/main/java/org/infinispan/commons/lambda/NamedBiConsumer.java]: I know it's probably more trouble than it's worth, but I'd like `NamedBiConsumer` to be nested in a `NamedLambdas` class with a `NamedLambdas.of(BiConsumer)` static method, so it's easier to add more functional interfaces later.

Comment (diegolovison) [reply commons-test/src/main/java/org/infinispan/commons/lambda/NamedBiConsumer.java]: @danberindei let me know what do you think

Comment (ryanemerson) [reply commons-test/src/main/java/org/infinispan/commons/lambda/NamedBiConsumer.java]: @diegolovison If you make `NamedBiConsumer` package-private or a non-public inner class of `NamedLambdas` we can guarantee that `NamedLambdas#of` will always be called. At the minute somebody writing a test could still do `new NamedBiConsumer`. Personally I prefer to have everything in `NamedLambdas`.

Comment (diegolovison) [reply commons-test/src/main/java/org/infinispan/commons/lambda/NamedBiConsumer.java]: @ryanemerson Done

**Comment by ryanemerson** [comment]
@diegolovison Can these changes be applied to other tests outside of core? (I haven't looked)

**Comment by diegolovison** [comment]
@ryanemerson I just picked random projects and tested. They are available.

**Comment by ryanemerson** [comment]
@diegolovison Right. So is the scope of this PR to replace all lambda paramters with `NamedLambdas.of`? Or are you just putting the capabilities in place for other devs to utilise in the future?

**Comment by diegolovison** [comment]
@ryanemerson both. I replaced all tests that were using Lambdas as a parameter in this PR. Basically, we don't need to replace all Lambdas, just that ones that we are using as a parameter in the test.

**Comment by danberindei** [comment]
Integrated, thanks Diego!

**Pull request merged**